frame/bounties/src/migrations/v4.rs

```diff
@@ -0,0 +1,134 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2019-2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use frame_support::{
+	traits::{Get, GetStorageVersion, STORAGE_VERSION_STORAGE_KEY_POSTFIX},
+	weights::Weight,
+};
+use sp_core::hexdisplay::HexDisplay;
+use sp_io::hashing::twox_128;
+
+/// Migrate the entire storage of this pallet to a new prefix.
+///
+/// This new prefix must be the same as the one set in construct_runtime. For safety, use
+/// `PalletInfo` to get it, as:
+/// `<Runtime as frame_system::Config>::PalletInfo::name::<BountiesPallet>`.
+///
+/// The migration will look into the storage version in order not to trigger a migration on an up
+/// to date storage. Thus the on chain storage version must be less than 4 in order to trigger the
+/// migration.
+pub fn migrate<T: frame_system::Config, P: GetStorageVersion, N: AsRef<str>>(
+	old_pallet_name: N,
+	new_pallet_name: N,
+) -> Weight {
+	let old_pallet_name = old_pallet_name.as_ref();
+	let new_pallet_name = new_pallet_name.as_ref();
+
+	if new_pallet_name == old_pallet_name {
+		log::info!(
+			target: "runtime::bounties",
+			"New pallet name is equal to the old prefix. No migration needs to be done.",
+		);
+		return 0
+	}
+
+	let on_chain_storage_version = <P as GetStorageVersion>::on_chain_storage_version();
+	log::info!(
+		target: "runtime::bounties",
+		"Running migration to v4 for bounties with storage version {:?}",
+		on_chain_storage_version,
+	);
+
+	if on_chain_storage_version < 4 {
+		log::info!(target: "runtime::bounties", "new prefix: {}", new_pallet_name);
+		frame_support::storage::migration::move_pallet(
+			old_pallet_name.as_bytes(),
+			new_pallet_name.as_bytes(),
+		);
```

Comment:
old pallet was stored at prefix `Treasury` which was shared with the treasury pallet, so the moving here will move both pallet-treasury storages and bounty storages (and tips storages).

Instead we should move individual storage using https://paritytech.github.io/substrate/frame_support/storage/migration/fn.move_storage_from_pallet.html

Reply:
you can look at https://github.com/paritytech/substrate/pull/9711 for an example